jukebox/src/utils.rs

```diff
@@ -10,3 +10,51 @@ pub(crate) fn copy_bytes_from_file(file: &mut File, offset: u64, size: usize) ->
     file.read_exact(&mut bytes).map_err(IsoRead)?;
     Ok(bytes)
 }
+
+/// Converts an HPS offset to its equivalent stage name
+/// Essentially, gets the stage belonging to a song
+pub(crate) fn hps_to_stage(offset: u64) -> Option<String> {
+    let stage = match offset {
+        // Legal stages
+        71324300 | 183962648 => "final_destination",
+        67606252 | 192481656 => "battlefield",
+        260526192 => "yoshis_story",
+        86968172 => "fountain_of_dreams",
+        135559532 => "dreamland",
+        156902380 | 160483852 => "pokemon_stadium",
+        // Misc
+        116194572 | 114173548 | 112087660 => "menu",
+        227529496 => "target_test",
+        // Other
+        258226448 | 180321356 => "yoshis_island",
+        138041836 => "yoshis_island_2",
+        129809196 => "kongo_jungle",
+        102329228 => "brinstar_depths",
+        40067628 => "fourside",
+        12767116 | 119550124 => "big_blue",
+        163012812 => "poke_floats",
+        79192940 | 24668300 => "kingdom",
+        83914380 => "kingdom_2",
+        264873488 => "brinstar",
+        144477996 | 140261388 => "onett",
+        124345132 => "mute_city",
+        168831468 => "rainbow_cruise",
+        94385004 => "jungle_japes",
+        173785356 => "temple",
+        56785644 => "green_greens",
+        250920912 => "venom",
+        10628844 | 75159404 => "icicle_mountain",
+        17422220 => "princess_peachs_castle",
+        45632108 => "kongo_jungle",
+        172460780 | 54558348 => "great_bay",
+        21447116 => "corneria",
+        36096908 => "flatzone",
+        _ => "",
```

Comment:
While this approach works and is something that was used in very early versions of jukebox, it can be a quite brittle. I highly encourage making use of what is known as the [FST \(Filesystem Table\)](https://www.gc-forever.com/yagcd/chap13.html#sec13.4). It's essentially a table of contents that lists every file on the disc, and it's offset in the ISO. Fortunately, I've already written code to work with this in an earlier version of Jukebox: https://github.com/project-slippi/slippi-rust-extensions/blob/ec8db587/jukebox/src/fst/mod.rs#L20-L89

In this code I walk the FST and produce a hashmap of `TrackID`s (an arbitrary enum) and the associated song's `offset` and `length` in bytes: `HashMap::<TrackId, (Offset, Length)>`

In your case, you would want to modify this so the hashmap that's produced is something like `HashMap::<Offset, StageDirPath>` where `Offset: usize` and `StageDirPath: std:path::Path`

```
{
  0x123456: Path("/path/to/User/Slippi/Music/yoshis_story"),
  0xABCDEF: Path("/path/to/User/Slippi/Music/final_destination"),
  ...
}
```

For example: you load into Yoshi's Story, you receive `0x123456` from `start_song()` and can look up the associated custom music directory in the hashmap with: 
```rs
let stage_dir = custom_song_path_map.get(0x123456);
```

Obviously the FST doesn't contain a list of _your_ custom folder paths in it (since we are making them up), but it _does_ contain the names of specific files. You will need an intermediary `match` that associates each `hps` filename with it's corresponding `stage_dir`:

```rs
match filename {
  "ystory.hps" => Some(Path::new("/path/to/User/Slippi/Music/yoshis_story")),
  "sp_end.hps" => Some(Path::new("/path/to/User/Slippi/Music/final_destination"))
  ...
  _ => None
}
```
You can use this `match` as a starting point: https://github.com/project-slippi/slippi-rust-extensions/blob/ec8db587/jukebox/src/tracks.rs#L59-L108

I know this FST stuff is a lot of esoteric mumbo jumbo, so feel free to message me on Discord and I'll be happy to help you along.